experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/node.ts

```diff
@@ -4,25 +4,66 @@
  */
 
 import { fail } from "../../../util";
+import { TreeNodeSchema } from "../../typed-schema";
+import { EditableTreeEvents } from "../../untypedTree";
 import { TreeNode } from "../editableTreeTypes";
-import { SharedTreeNode } from "./types";
-
-export const nodeSym = Symbol("node");
+import { getProxyForNode } from "./proxies";
+import { ProxyNode, SharedTreeNode, getTreeNode } from "./types";
 
 /**
- * Exposes node functionality, such as event subscriptions, for the given SharedTreeNode.
+ * The `node` object holds various functions for analyzing {@link SharedTreeNode}s.
  * @alpha
  */
-export function node(owner: SharedTreeNode): Required<SharedTreeNode>[typeof nodeSym] {
-	return (
-		(owner[nodeSym] as Required<SharedTreeNode>[typeof nodeSym]) ??
-		fail("owner must be a SharedTreeNode")
-	);
-}
+export const node = {
+	/**
+	 * The schema information for this node.
+	 */
+	schema: (n: SharedTreeNode) => {
+		return assertTreeNode(n).schema;
+	},
```

Comment:
Nit: I'm guessing you chose 'n' to avoid shadowing the outer node variable?

It's not terrible, but I personally would suppress lint and use 'node'.